### ISSUE TITLE
iOS Volume

### ISSUE THREAD

**Issue by ElaineSchwner** [dlutton/flutter_tts#106]
## 🐛 Bug Report

Audio in IOS is only from the telephone speakers, not outloud speakers.

The same code in IOS is from the speakers.

**Comment by dlutton** [comment]
@ElaineSchwner sorry I don't follow, can you be a little more descriptive of the issue?

**Comment by ElaineSchwner** [comment]
@dlutton Thank you for your attention.
The same code, In Android, the sound is very good and clear.
In iOS, the sound is very low, almost inaudible.

**Comment by dlutton** [comment]
Okay.  This PR #103 should fix that issue for you.  I should have it reviewed and merged by this weekend along with some other bug fixes.

**Comment by ElaineSchwner** [comment]
@dlutton Ok. Thank you. Do I have to do anything to update my code? or wait? What is the next step?
Thank you.

**Comment by dlutton** [comment]
For now you'll have to wait.  Once the PR is merged, you'll need to update to the latest flutter_tts package in your pubspec.yaml file and implement the .setIosAudioCategory method.  This is only for iOS, so you'll want to make sure you're checking if the platform is iOS before calling that method.

**Comment by dlutton** [comment]
@ElaineSchwner the latest update has been published.

**Comment by ElaineSchwner** [comment]
@dlutton  Thank you Daniel. The sound now is from the speakers, but the .setSpeechRate can't be set. Do you have any idea why it is happening and how it can be fixed?

**Comment by dlutton** [comment]
@ElaineSchwner do you mind opening a new issue so I can track it.  Thank you.

**Comment by siva636** [comment]
This does not work for me. The audio is very low sometimes and it is as expected sometimes. This only works about 50% of the playback. Automatically audio volume goes to low. I tried with the following code:

```
await tts.setIosAudioCategory(IosTextToSpeechAudioCategory.playback,
        [IosTextToSpeechAudioCategoryOptions.defaultToSpeaker]);
    await tts.setIosAudioCategory(IosTextToSpeechAudioCategory.playAndRecord,
        [IosTextToSpeechAudioCategoryOptions.defaultToSpeaker]);
```

**Comment by siva636** [comment]
Can this thread be reopened or I have to file a new issue?

**Comment by siva636** [comment]
@ dlutton Yes I tested using the latest version ^1.2.1.

Sometimes the volume goes to the lowest level / highest level without any user interaction. This happens only on ios physical device. Not happening on ios/android simulators.
(I did not test on physical android device)

**Comment by seanradford** [comment]
We're using flutter_tts:1.2.6 (current latest) and seeing this exact issue. On iOS, the majority of the time (think 95%) the volume is so low to be almost inaudible (phone volume at max and flutter_tts volume set to 1.0).

**Comment by seanradford** [comment]
As an addendum to my previous post. It does appear that with headphones the volume is audible 'all or the time'.

And to re-iterate, this is with 1.2.6 (latest version). As such should this issue be re-opened?

**Comment by dlutton** [comment]
thanks for reporting this issue @seanradford, I'll take a gander.

**Comment by r0hit-gupta** [comment]
I am facing the same issue. The volume is almost inaudible for most iOS devices.

**Comment by seanradford** [comment]
Hi @r0hit-gupta,

Try adding the defaultToSpeaker option and see if that helps:

```
if (Platform.isIOS) {
  await _flutterTts
    .setIosAudioCategory(IosTextToSpeechAudioCategory.playback, [
    IosTextToSpeechAudioCategoryOptions.allowBluetooth,
    IosTextToSpeechAudioCategoryOptions.allowBluetoothA2DP,
    IosTextToSpeechAudioCategoryOptions.mixWithOthers,
    IosTextToSpeechAudioCategoryOptions.defaultToSpeaker
  ]);
}
```

**Comment by r0hit-gupta** [comment]
@seanradford thanks a lot. It worked for me.
